prov/efa/src/efa_hmem.c

```diff
@@ -276,70 +227,79 @@ static int efa_domain_hmem_info_init_neuron(struct efa_domain *efa_domain)
 		EFA_WARN(FI_LOG_DOMAIN,
 			 "Failed to deregister Neuron buffer: %s\n",
 			 fi_strerror(-ret));
-		return ret;
+		return;
 	}
 
 	info->p2p_supported_by_device = true;
-	efa_domain_hmem_info_init_protocol_thresholds(efa_domain, FI_HMEM_NEURON);
-	if (-FI_ENODATA != fi_param_get(&efa_prov, "inter_max_medium_message_size", &tmp_value)) {
-		EFA_WARN(FI_LOG_DOMAIN,
-		         "The environment variable FI_EFA_INTER_MAX_MEDIUM_MESSAGE_SIZE was set, "
-		         "but EFA HMEM via Neuron API only supports eager and runting read protocols. "
-				 "The variable will not modify Neuron memory run config.\n");
-	}
-
+	return;
 #endif
-	return 0;
+	return;
+}
+
+static inline bool efa_domain_hmem_info_is_p2p_supported_by_device(struct efa_domain *efa_domain, enum fi_hmem_iface iface) {
```

Comment:
This function name gives reader an impression that it can be called multiple times. But I do want it is a one-time call. So you either rename the function make it a light weight call that only return the `info->p2p_supported_by_device` related stuff, and do the heavy-weight check call separately.

Reply:
Why would we call this functions multiple times? We would just get info->p2p_supported_by_device right?

Reply:
In convention, `is_p2p_supported_by_device`  should be a simple lightweight boolean get, and it is expected that we can call this function cheaply to get device support status. I don't want the heavy check code as part of it.